lib/HTFeed/JobMetrics.pm

```diff
@@ -318,61 +360,74 @@ sub _setup_metrics {
     # Prefix all metric names with:
     my $pfx = "ingest";
 
-    # Each element x in this array gets turned into 3 prom declarations:
-    # x_seconds, x_bytes and x_items
-    # Names are based on the downcased class name of the stage.
+    # Each element in @measurable_stages gets combined with
+    # each element in @units, to produce e.g. pack_items etc.
+    # Names are based on the downcased class name of the stage,
+    # or method in which the measured event happens.
     my @measurable_stages = (
-	'collate',
-	'download',
-	'download_dropbox',
-	'download_google',
-	'download_ia',
-	'handle',
-	'imageremediate',
-	'mets',
-	'pack',
-	'sourcemets',
-	'unpack',
-	'verifymanifest',
-	'volumevalidator'
+        'collate',
+        'download',
+        'download_dropbox',
+        'download_google',
+        'download_ia',
+        'handle',
+        'imageremediate',
+        'encrypt',
+        'mets',
+        'move',
+        'pack',
+        'postvalidate',
+        'prevalidate',
+        'record_audit',
+        'record_backup',
+        'sourcemets',
+        'stage',
+        'unpack',
+        'validate_zip_completeness',
+        'verify_crypt',
+        'verifymanifest',
+        'volumevalidator',
+    );
+    my @units = (
+        'bytes_read',
+        'bytes_write',
```

Comment:
This naming convention is difficult. I would propose the alternative "bytes_read" and "bytes_written" although it seems ungainly. Alternatives to that: [rw]bytes, bytes[rw], bytes_[in,out], [input,output]_bytes. The main thing I am looking for is for the last element of the metric to be the "thing measured", just trying to work around the fact that two irregular verbs are not fun to contend with.

Reply:
Looking at existing metrics it should probably end in `bytes_total` or maybe just `bytes` is fine. A couple examples I see from existing metrics:

* container_fs_reads_bytes_total
* container_fs_writes_bytes_total
* node_disk_read_bytes_total
* node_disk_written_bytes_total
* ceph_osd_op_r_out_bytes
* ceph_osd_op_w_in_bytes

I believe the `_total` is to indicate that it's a counter rather than a gauge (e.g. to differentiate it from something like `container_fs_usage_bytes`) -- but the ceph metrics are also counters and don't end in `_total` so 🤷

Reply:
I'm leaning towards `_r_` and `_w_` as these strings are silly long already. And speaking of long, I'm adding `_total` to all current metric names (because they are all counters).